modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,491 @@ redirect_from:
+  if (s.matches(Regex("-?[0-9]+"))) Either.Right(s.toInt())
+  else Either.Left(NumberFormatException("$s is not a valid integer."))
+
+fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+  if (s.matches(Regex("-?[0-9]+"))) s.toInt().validNel()
+  else NumberFormatException("$s is not a valid integer.").invalidNel()
+```
+We can now `traverse` structures that contain strings parsing them into integers and accumulating failures with `Either`.
+
+```kotlin:ank:playground
+import arrow.core.Either
+import arrow.core.extensions.either.applicative.applicative
+import arrow.core.k
+import arrow.core.ValidatedNel
+import arrow.core.extensions.either.foldable.isEmpty
+import arrow.core.invalidNel
+import arrow.core.validNel
+
+fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+  if (s.matches(Regex("-?[0-9]+"))) Either.Right(s.toInt())
+  else Either.Left(NumberFormatException("$s is not a valid integer."))
+
+fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+  if (s.matches(Regex("-?[0-9]+"))) s.toInt().validNel()
+  else NumberFormatException("$s is not a valid integer.").invalidNel()
+
+fun main() {
+  //sampleStart
+  val list = listOf("1", "2", "3").k().traverse(Either.applicative(), ::parseIntEither)
+  val isLeft = listOf("1", "abc", "3").k().traverse(Either.applicative(), ::parseIntEither).isEmpty()
+  //sampleEnd
+  println("list= $list")
+  println("isLeft= $isLeft")
+}
+```
+
+```kotlin:ank:playground
+import arrow.core.Either
+import arrow.core.Nel
+import arrow.core.ValidatedNel
+import arrow.core.extensions.nonemptylist.semigroup.semigroup
+import arrow.core.extensions.validated.applicative.applicative
+import arrow.core.fix
+import arrow.core.invalidNel
+import arrow.core.k
+import arrow.core.validNel
+
+fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+  if (s.matches(Regex("-?[0-9]+"))) Either.Right(s.toInt())
+  else Either.Left(NumberFormatException("$s is not a valid integer."))
+
+fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+  if (s.matches(Regex("-?[0-9]+"))) s.toInt().validNel()
+  else NumberFormatException("$s is not a valid integer.").invalidNel()
+
+fun main() {
+  //sampleStart
+  val validatedList = listOf("1", "22w", "3", "33s").k()
+    .traverse(ValidatedNel.applicative(Nel.semigroup<NumberFormatException>()), ::parseIntValidated).fix()
+  //sampleEnd
+  println(validatedList)
+}
+```
+
+Notice that in the `Either` case, should any string fail to parse the entire `traversal` is considered a failure. Moreover, once it hits its first bad parse, it will not attempt to parse any others down the line (similar behavior would be found with using `Option`). Contrast this with `Validated` where even if one bad parse is hit, it will continue trying to parse the others, accumulating any and all errors as it goes. The behavior of traversal is closely tied with the `Applicative` behavior of the data type.
```

Comment:
```suggestion
Notice that in the `Either` case, should any string fail to parse the entire `traverse` is considered a failure. Moreover, once it hits its first bad parse, it will not attempt to parse any others down the line (similar behavior would be found with using `Option`). Contrast this with `Validated` where even if one bad parse is hit, it will continue trying to parse the others, accumulating any and all errors as it goes. The behavior of traversal is closely tied with the `Applicative` behavior of the data type.
```